### ISSUE TITLE
BRPOPLPUSH never returns if key doesn't exist when it was called

### ISSUE THREAD

**Issue by weisserw** [alicebob/miniredis#97]
To reproduce:

```
import (
    "fmt"
    "github.com/alicebob/miniredis"
    "github.com/go-redis/redis"
    "time"
)

var addr string

func main() {
    fakeredis, err := miniredis.Run()
    if err != nil {
        panic(err)
    }

    addr = fakeredis.Addr()

    fakeredis.Lpush("queue1", "already existing value")

    go readVals()

    time.Sleep(1 * time.Second)

    fakeredis.Lpush("queue1", "new value")

    for {
    }
}

func readVals() {
    rdb := redis.NewClient(&redis.Options{Addr: addr})
    rdb.BRPopLPush("queue1", "queue2", 0)
    fmt.Println("This works OK")
    rdb.BRPopLPush("queue1", "queue2", 0)
    fmt.Println("Never gets here :-(")
}
```

readVals calls BRPopLPush which returns "already existing value", then calls BRPopLPush again, which blocks because the key is now empty. One second later in main another value is pushed. I expect BRPopLPush to return a second time but it does not.

**Comment by alicebob** [comment]
thanks for the bug report! I'll have a look soon.

**Comment by alicebob** [comment]
@weisserw, it should be fixed in #98. I'll merge that in a few days, or you can use the branch directly.

Thanks for the clear report!

**Comment by alicebob** [comment]
merged into master, thanks!
